### ISSUE TITLE
Zed: Outdated diagnostics when using `diagnosticMode: "openFiles"`

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ty#1622]
Create a file `lib.py`:

```py
class Project:
    name: str

```

Create a second file `main.py`:

```py
from lib import Project

p = Project()

print(p.name)
```

Now, change `lib.py` to 

```py
class Project:
    name: str

```

There should now be a diagnostic for `print(p.name)` but Zed fails to fetch the diagnostics for `main.py`. 

Note, that ty uses `relatedInformation: true` when registering for pull diagnostics.

I reported this issue upstream, see https://github.com/zed-industries/zed/issues/31259#issuecomment-3571961820
